queue_services/activity-log-listener/q_cli.py

```diff
@@ -0,0 +1,176 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service for listening and handling Queue Messages.
+
+This service registers interest in listening to a Queue and processing received messages.
+"""
+import asyncio
+import functools
+import getopt
+import json
+import os
+import random
+import signal
+import sys
+from datetime import datetime
+
+from nats.aio.client import Client as NATS  # noqa N814; by convention the name is NATS
+from stan.aio.client import Client as STAN  # noqa N814; by convention the name is STAN
+
+from entity_queue_common.service_utils import error_cb, logger, signal_handler
+
+
+async def run(loop, mode, auth_account_id, auth_account_name, bank_number, bank_branch_number,
+              bank_account_number, order_number, transaction_amount, transaction_id):  # pylint: disable=too-many-locals
+    """Run the main application loop for the service.
+
+    This runs the main top level service functions for working with the Queue.
+    """
+    # NATS client connections
+    nc = NATS()
+    sc = STAN()
+
+    async def close():
+        """Close the stream and nats connections."""
+        await sc.close()
+        await nc.close()
+
+    # Connection and Queue configuration.
+    def nats_connection_options():
+        return {
+            'servers': os.getenv('NATS_SERVERS', 'nats://127.0.0.1:4222').split(','),
+            'io_loop': loop,
+            'error_cb': error_cb,
+            'name': os.getenv('NATS_CLIENT_NAME', 'entity.filing.tester')
+        }
+
+    def stan_connection_options():
+        return {
+            'cluster_id': os.getenv('NATS_CLUSTER_ID', 'test-cluster'),
+            'client_id': str(random.SystemRandom().getrandbits(0x58)),
+            'nats': nc
+        }
+
+    def subscription_options():
+        return {
+            'subject': os.getenv('NATS_SUBJECT', 'account.events'),
+            'queue': os.getenv('NATS_QUEUE', 'account.events.worker'),
+            'durable_name': os.getenv('NATS_QUEUE', 'account.events.worker') + '_durable'
+        }
+
+    try:
+        # Connect to the NATS server, and then use that for the streaming connection.
+        await nc.connect(**nats_connection_options())
+        await sc.connect(**stan_connection_options())
+
+        # register the signal handler
+        for sig in ('SIGINT', 'SIGTERM'):
+            loop.add_signal_handler(getattr(signal, sig),
+                                    functools.partial(signal_handler, sig_loop=loop, sig_nc=nc, task=close)
+                                    )
+        payload = None
+        if mode == 'lock':
```

Comment:
this file need changes specific to activity log q messages.

Reply:
done